New GA criteria. Rename GA to Stable. (#1880)
dapr/docs: daprdocs/content/en/operations/components/certification-lifecycle.md

- The component contains a record of the conformance test result reviewed and approved by Dapr maintainers with specific components-contrib version
- Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases

### Stable

- The component must have component [certification tests](#certification-tests) validating functionality and resiliency
- The component is maintained by Dapr maintainers and supported by the community
- The component is well documented and tested

> - A maintainer will address component security, core functionality and test issues according to the Dapr support policy and issue a patch release of Dapr that includes the patched stable component

Reviewer: ```suggestion
- A maintainer will address component security, core functionality and test issues according to the Dapr support policy and issue a patch release that includes the patched stable component
```

> ### Previous General Available (GA) components

Reviewer 2: Either `Generally Available` or `General Availability`

Author: Done.

> Any component that was previously certified as GA is now grandfathered into Stable even if the new requirements are not met.

Reviewer 3: Adjusting for more inclusive term.

```suggestion
Any component that was previously certified as GA is allowed into Stable even if the new requirements are not met.
```

> Any component that was previously certified as GA is now grandfathered into Stable even if the new requirements are not met yet.

Reviewer: ```suggestion
Any component that was previously certified as GA is now grandfathered into Stable even if the new requirements are not met yet.
```
```suggestion
Any component that was previously certified as GA is now grandfathered into Stable even if the new requirements are not met yet.
```
```suggestion
Any component that was previously certified as GA is now grandfathered into Stable even if the new requirements are not met.
```

- The tests should validate the functional behavior and robustness of component based on the component specification
- All the details needed to reproduce the tests are added as part of the component conformance test documentation

## Certification tests

Each stable component in the [components-contrib](https://github.com/dapr/components-contrib) repository must have a certification test plan and automated certification tests validating all features supported by the component via Dapr.

Test plan for stable components should include the following scenarios:

> - Client reconnection: in case the client library cannot connect to the service for a moment, Dapr sidecar should not require to be restarted once the service is back online.

Reviewer: ```suggestion
- Client reconnection: in case the client library cannot connect to the service for a moment, Dapr sidecar should not require a restart once the service is back online.
```

Reviewer 4: Note: There is no Dapr team only Dapr project

- Authentication options: validate the component can authenticate with all the supported options.
- Validate resource provisioning: validate if the component automatically provisions resources on initialization, if applicable.
- All scenarios relevant to the corresponding building block and component.

The test plan must be approved by a Dapr maintainer and be published in a `README.md` file along with the component code.

### Test requirements

- The tests should validate the functional behavior and robustness of the component based on the component specification, reflecting the scenarios from the test plan

> - The tests must run successfully as part of the continuous integration of the `components-contrib` repository

Reviewer 3: ```suggestion
- The tests must run successfully as part of the continuous integration of the [components-contrib](https://github.com/dapr/components-contrib) repository
```

> - The tests should validate the functional behavior and robustness of component based on the component specification, reflecting the scenarios from the test plan

Reviewer: ```suggestion
- The tests should validate the functional behavior and robustness of the component based on the component specification, reflecting the scenarios from the test plan
```

> - The tests must run successfully as part of the continuous integration of `components-contrib` repository

Reviewer: ```suggestion
- The tests must run successfully as part of the continuous integration of the `components-contrib` repository
```

## Component certification process

> For a component to be certified tests are run in an environment maintained by the Dapr team.

Reviewer: ```suggestion
In order for a component to be certified, tests are run in an environment maintained by the Dapr project.
```

Reviewer 4: There is no Dapr team i.e. this "by the Dapr team". 

Change to 
"In order for a component to be certified, tests are run in an environment maintained by the Dapr project."

### New component certification: Alpha->Beta

For a new component requiring a certification change from Alpha to Beta, a request for component certification follows these steps:

> - Requestor creates an issue in `components-contrib` repository for certification of the component with the current and the new certification levels

Reviewer: ```suggestion
- Requestor creates an issue in the `components-contrib` repository for certification of the component with the current and the new certification levels
```

> - Requestor creates an issue in the `components-contrib` repository for certification of the component with the current and the new certification levels

Reviewer 3: ```suggestion
- Requestor creates an issue in the [components-contrib](https://github.com/dapr/components-contrib) repository for certification of the component with the current and the new certification levels
```

> - Requestor submits a PR for integrating the component to run with the defined conformance test suite, if not included already

Reviewer: ```suggestion
- Requestor submits a PR to integrate the component with the defined conformance test suite, if not already included
```

>     - The user details the environment setup in the issue created, so that a Dapr maintainer can setup the service in a managed environment

Reviewer: ```suggestion
    - The user details the environment setup in the issue created, so a Dapr maintainer can setup the service in a managed environment
```

    - After the environment setup is complete, Dapr maintainers review the PR and if approved merges that PR

> - Requestor submits a PR in `docs` repository, updating the component's certification level

Reviewer: ```suggestion
- Requestor submits a PR in the `docs` repository, updating the component's certification level
```

> - Requestor submits a PR in the `docs` repository, updating the component's certification level

Reviewer 3: ```suggestion
- Requestor submits a PR in the [docs](https://github.com/dapr/docs) repository, updating the component's certification level
```

### New component certification: Beta->Stable

For a new component requiring a certification change from Beta to Stable, a request for component certification follows these steps:

> - Requestor creates an issue in `components-contrib` repository for certification of the component with the current and the new certification levels

Reviewer: ```suggestion
- Requestor creates an issue in the `components-contrib` repository for certification of the component with the current and the new certification levels
```

> - Requestor creates an issue in the `components-contrib` repository for certification of the component with the current and the new certification levels

Reviewer 3: ```suggestion
- Requestor creates an issue in the [components-contrib](https://github.com/dapr/components-contrib) repository for certification of the component with the current and the new certification levels
```

- Requestor submits a PR for the test plan as a `README.md` file in the component's source code directory

>     - The user details the test environment requirements in the PR created, including any manual steps or credentials needed

Reviewer 3: ```suggestion
    - The requestor details the test environment requirements in the created PR, including any manual steps or credentials needed
```

>     - A Dapr maintainer reviews the test plan, giving feedback or approving it and merge the PR

Reviewer: ```suggestion
    - A Dapr maintainer reviews the test plan, giving feedback or approving it and merges the PR
```

>     - A Dapr maintainer reviews the test plan, giving feedback or approving it and merges the PR

Reviewer 3: ```suggestion
    - A Dapr maintainer reviews the test plan, provides feedback or approves it, and eventually merges the PR
```

- Requestor submits a PR for the automated certification tests, including scripts to provision resources when applicable
- After the test environment setup is completed and credentials provisioned, Dapr maintainers review the PR and, if approved, merges the PR

> - Requestor submits a PR in `docs` repository, updating the component's certification level

Reviewer: ```suggestion
- Requestor submits a PR in the `docs` repository, updating the component's certification level
```

> - Requestor submits a PR in the `docs` repository, updating the component's certification level

Reviewer 3: ```suggestion
- Requestor submits a PR in the [docs](https://github.com/dapr/docs) repository, updating the component's certification level
```
